Lib/logging/__init__.py

```diff
@@ -1506,17 +1507,11 @@ def findCaller(self, stack_info=False, stacklevel=1):
         Find the stack frame of the caller so that we can note the source
         file name, line number and function name.
         """
-        f = currentframe()
+        f = currentframe(self.default_stack_level + stacklevel)
```

Comment:
Shouldn't the unwinding really be done later? A few lines down, we read:
```python
        while hasattr(f, "f_code"):
            co = f.f_code
            filename = os.path.normcase(co.co_filename)
            if filename == _srcfile:
                f = f.f_back
                continue
```
This 'escapes' the stack frames generated inside the current file. I would expect any additional `stacklevel` treatment to happen afterwards.

As a result of the current implementation (with or without the proposed changes in the current PR), the `stacklevel` parameter to `.warn` must be one higher than the parameter to `.warning` in order to get to the same frame.